source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -231,6 +229,13 @@ bool DnsMessageParser::parseDnsObject(DnsQueryContextPtr& context,
     }
     context->queries_.push_back(std::move(rec));
   }
+
+  // We could encounter additional RRs in an EDNS query, and Envoy will ignore them.
+  if (context->header_.additional_rrs) {
+    ENVOY_LOG(debug, "Ignoring additional RRs in a query because Envoy does not support. "
```

Comment:
Should we add a counter here, then in the test check the counter is increased if such query is received?

Reply:
Done! Thank you!